### ISSUE TITLE
Godot 3.0.3 RC1 Mono:  Cannot find class <ClassName> for script

### ISSUE THREAD

**Issue by insraq** [godotengine/godot#18584]
**Godot version:**
Godot 3.0.3 RC1

**OS/device including version:**
macOS 10.13.4
Mono 5.8.1.0

**Issue description:**
The same project works in 3.0.2. But in 3.0.3, after opening the projet, the editor prints the error: `Cannot find class <ClassName> for script.`. And when running the project, the scripts do not run either.

**Minimal reproduction project:**
Here's the repository: https://github.com/insraq/traffic_planner

**Comment by insraq** [comment]
I have tried to build the same project on Windows, the project does not even compile :-(

**Comment by insraq** [comment]
This is the build error on Windows

```
System.ComponentModel.Win32Exception: ApplicationName='C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe', CommandLine='"C:\Users\***\AppData\Roaming/Godot/mono/solutions/13097970639814129045_2_2/GodotSharp/GodotSharp.sln" /v:normal /t:Build "/p:Configuration=Release" "/l:GodotSharpTools.Build.GodotBuildLogger,C:\Users\***\Projects\Godot_v3.0.3-rc1_mono_win64/GodotSharpTools.dll;C:\Users\***\AppData\Roaming/Godot/mono/build_logs/c9c1d4d4114a17c0eb783d050afcc195_Release" "/p:NoWarn=1591" "/p:FrameworkPathOverride=C:\Program Files (x86)\Mono\lib\/mono/4.5"', CurrentDirectory='', Native error= The operation completed successfully.
```

I've checked the log folder `C:\Users\***\AppData\Roaming/Godot/mono/build_logs/c9c1d4d4114a17c0eb783d050afcc195_Release`, it does not exist.

**Comment by KellyThomas** [comment]
This command includes the parameter `"/p:FrameworkPathOverride=C:\Program Files (x86)\Mono\lib\/mono/4.5"'`.

Are you running mono 5.10.x on your windows machine?

**Comment by hpvb** [comment]
There were some issues with the 3.0.3-rc1 build. 3.0.3-rc2 is out now, could you please try again?

**Comment by insraq** [comment]
I can confirm it works with 3.0.3-rc2 now. However, exporting to Windows still does not work. Here's the error message.
![snipaste_2018-05-15_00-46-37](https://user-images.githubusercontent.com/608221/40025209-89ad0cec-57d9-11e8-996f-fe3fa2f8b127.png)

**Comment by hpvb** [comment]
for 3.0.3 I'll add a little workaround to ship the mono-2.0-sgen.dll with the game. For 3.0.4 it should be statically linked again.

**Comment by hpvb** [comment]
This was fixed in 3.0.4.
